src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -766,10 +832,10 @@ private static int WriteAsTimestamp(DateTimeOffset timestamp, Span<byte> destina
             return FormatOctal(unixTimeSeconds, destination);
         }
 
-        // Writes the specified text as an ASCII string aligned to the left, and returns its checksum.
-        private static int WriteAsAsciiString(string str, Span<byte> buffer)
+        // Writes the specified text as an UTF8 string aligned to the left, and returns its checksum.
+        private static int WriteAsUtf8String(string str, Span<byte> buffer)
         {
-            byte[] bytes = Encoding.ASCII.GetBytes(str);
+            byte[] bytes = Encoding.UTF8.GetBytes(str);
             return WriteLeftAlignedBytesAndGetChecksum(bytes.AsSpan(), buffer);
```

Comment:
I don't remember why I didn't fix this as part of doing a perf pass over all this code (maybe this is rarely used and so it wasn't popping in traces?), but subsequent to this change, if this is possibly hot path, we should fix this function to avoid allocating a byte[] on every call.  Presumably in many circumstances it could even encode directly into the output buffer.